microcosm_caching/decorators.py

```diff
@@ -118,3 +126,73 @@ def cache(*args, **kwargs) -> Schema:
 
         return cache
     return decorator
+
+
+def invalidates(
+    component,
+    invalidations,
+    cache_prefix,
+    lock_ttl=DEFAULT_LOCK_TTL
+):
+    """
+    Invalidates a set of prescribed keys, based on a combination of:
+        * specified arguments
+        * schema
+
+    Note: this does require that the input args to the given function correspond to the
+    invalidation args given. As such, using it conjunction with certain functions (such as
+    a plain delete) may not provide sufficient context for invalidation, and caching should
+    not be used in such scenarios.
+
+    """
+    graph = component.graph
+    metrics = get_metrics(graph)
+    resource_cache: CacheBase = graph.resource_cache
+
+    def delete_from_cache(values) -> None:
+        """
+        "Delete" from cache by locking writes to a key for a designated
```

Comment:
why are you setting a short timeout rather than deleting here?

Reply:
So.

If you delete, you have to deal with interleaving, which I think I pointed out earlier. I tried `cas`, but it didn't appear to do what I needed it to do. That would work in a world where I was interleaving a replace type of operation (updating a resource in place), but that wasn't the path I was going down here.

This is essentially marking a resource as "bypass cache", as the cached decorator will now always fallback to the main function (since this always returns `None`), at least until the TTL expires. It's essentially like a "write" flag, but with less of a dance.

Reply:
I get it, but let's revisit later.